wormhole/file_transport.go

```diff
@@ -184,40 +184,73 @@ type fileTransport struct {
 	appID           string
 }
 
-func (t *fileTransport) connectViaRelay(otherTransit *transitMsg) (net.Conn, error) {
+// combines hints and remove duplicates to minimaze connections
+func appendAndRemoveDuplicateHints(myTransit []transitHintsV1, peerTransit []transitHintsV1, hintType string) []transitHintsRelay {
+	var combinedHints []transitHintsRelay
+
+	for _, peerHints := range peerTransit {
+		if peerHints.Type == hintType {
+			combinedHints = peerHints.Hints
+
+		}
+
+	}
+
+	for _, myHints := range myTransit {
+		if myHints.Type == hintType {
+			for _, myHint := range myHints.Hints {
+				var duplicate bool = false
+				for _, peerHint := range combinedHints {
+					if myHint == peerHint {
+						duplicate = true
+					}
+				}
+				if !duplicate {
+					combinedHints = append(combinedHints, myHint)
+				}
+			}
+		}
+
+	}
+	return combinedHints
+}
+
+func (t *fileTransport) connectViaRelay(myTransit *transitMsg, peerTransit *transitMsg) (net.Conn, error) {
 
 	successChan := make(chan net.Conn)
 	failChan := make(chan string)
 
 	var count int
 
-	for _, relay := range otherTransit.HintsV1 {
-		if relay.Type == "relay-v1" {
-			for _, endpoint := range relay.Hints {
-				var relayUrl *url.URL
-				var err error
-
-				switch endpoint.Type {
-				case "direct-tcp-v1":
-					relayUrl = &url.URL{
-						Scheme: "tcp",
-						Host:   net.JoinHostPort(endpoint.Hostname, strconv.Itoa(endpoint.Port)),
-					}
-				case "websocket-v1":
-					relayUrl, err = url.Parse(endpoint.Url)
+	var cancelArray []context.CancelFunc
 
-				}
-				ctx, cancel := context.WithCancel(context.Background())
+	combinedHints := appendAndRemoveDuplicateHints(myTransit.HintsV1, peerTransit.HintsV1, "relay-v1")
```

Comment:
It looks like we are not connecting to the direct hints anymore. Why?

Reply:
We connect either direct-tcp-v1 either websocket-v1 (line: 234-239) or I didn't get comment.